### PR TITLE
Fix #224 Update to electron v14

### LINKED ISSUE #224: Update to a supported electron version
We previously had blocking dependencies that limited us to v9 but that should no longer be the case. Update to a supported version of electron where everything still works.

Note there is an existing bot PR #221 that could be an interim step - checkout the branch and check if things still work.

### PR DESCRIPTION
**Pull request by irahopkinson**
Finally back to the later version of electron.

`npm` vulnerabilities have actually gotten worse since the upgrade from electron v9 to v11 already on master but improve with this update. The main issue with vulnerabilities now is that we are still using `@rescripts/cli` (deprecated and archived) to run and build.

### CODE REVIEW

File: main/electron.ts [hunk 1]
```diff
@@ -46,11 +46,11 @@ export function createWindow(): void {
...
   });
-  mainWindow.webContents.on('new-window', (event: Event, url: string): void => {
-    if (url.startsWith('http:') || url.startsWith('https:')) {
-      event.preventDefault();
-      shell.openExternal(url);
+  mainWindow.webContents.setWindowOpenHandler((details: HandlerDetails) => {
+    if (details.url.startsWith('http:') || details.url.startsWith('https:')) {
+      shell.openExternal(details.url);
     }
+    return { action: 'deny' };
```
Comment (JCFarrow) [inline main/electron.ts:53]: This part doesn't seem to be working correctly. A command prompt window opens briefly when starting and finishing rendering and when closing the program after rendering. I found this issue talking about it:
https://github.com/electron/electron/issues/27967

**Review by JCFarrow — CHANGES_REQUESTED** [review]
Looking good. It's just the `WindowOpenHandler` in `electron.ts` isn't working properly.

File: main/electron.ts (hunk 1 above)
Comment (irahopkinson) [reply main/electron.ts]: It wasn't `app.openPath` that was causing the console windows but the calls to spawn Ffmpeg.
I've fixed everything apart from the one when closing the app after a render.

**Comment by JCFarrow** [comment]
If this pull request makes npm vulnerabilities worse and it introduces an unnecessary window popup on close, then is it worth merging? Are the benefits better than these two problems?

**Comment by irahopkinson** [comment]
@JCFarrow It actually reduces the vulnerabilities over master (going from electron v9 to v11 made it worse on master). The user-facing vulnerabilities should go down as we move to the latest electron. Its the developer-facing vulnerabilities that are getting worse because we are still using `@rescripts/cli`.

**Comment by irahopkinson** [comment]
I've back off to electron v14  (rather than the latest v16 LTS) and the CMD window is not showing anymore.

**Review by JCFarrow — COMMENTED** [review]
It's good that the window isn't popping up anymore. Can/Should the `windowsHide: true` be removed since it doesn't look like it's needed now? I'm also getting the following two console errors:
```
[18260:1216/145438.437:ERROR:CONSOLE(168)] "Electron sandbox_bundle.js script failed to run", source: electron/js2c/sandbox_bundle.js (168)
[18260:1216/145438.437:ERROR:CONSOLE(168)] "TypeError: object null is not iterable (cannot read property Symbol(Symbol.iterator))", source: electron/js2c/sandbox_bundle.js (168)
```

**Comment by irahopkinson** [comment]
I'm inclined to keep `windowsHide: true` since they did help after going to v16 so we will need them again later and they don't do any harm now. Thanks for the reminder about the console errors - I did notice them but forgot to follow-up on them.

**Comment by irahopkinson** [comment]
The console errors above are resolved. 
Then I added a commit to fix the GHA build. Its a little hacky to add a _devDependency_ to fix this but it actually reduces our dependency tree as both `url-parse-lax` further down in the tree are now deduped to the one at the root.

**Review by JCFarrow — APPROVED** [review]
Looks good 👍

**Pull request merged**